lib/events.js

```diff
@@ -212,22 +155,13 @@ EventEmitter.prototype.emit = function emit(type, ...args) {
     needDomainExit = true;
   }
 
-  const isFn = typeof handler === 'function';
-  switch (args.length) {
-    case 0:
-      emitNone(handler, isFn, this);
-      break;
-    case 1:
-      emitOne(handler, isFn, this, args[0]);
-      break;
-    case 2:
-      emitTwo(handler, isFn, this, args[0], args[1]);
-      break;
-    case 3:
-      emitThree(handler, isFn, this, args[0], args[1], args[2]);
-      break;
-    default:
-      emitMany(handler, isFn, this, args);
+  if (typeof handler === 'function') {
+    handler.apply(this, args);
+  } else {
+    const len = handler.length;
+    const listeners = arrayClone(handler, len);
```

Comment:
Can you leave a TODO here to consider switching to `Array.prototype.slice` once V8 6.4 lands in Node?